compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -353,6 +385,31 @@ LogicalResult MapScatterOp::verify() {
   return success();
 }
 
+bool MapScatterOp::isIdentity() {
+  if (getInputType() != getOutputType()) {
+    return false;
+  }
+
+  // Check that the mask is always true.
+  Block &transformBody = getTransformationRegion().getBlocks().front();
+  auto yieldOp = cast<IREE::LinalgExt::YieldOp>(transformBody.getTerminator());
+  Value mask = yieldOp->getOperands().back();
+  std::optional<int64_t> constMask = getConstantIntValue(mask);
+  if (!constMask.has_value() || constMask.value() == 0) {
```

Comment:
I learned that you can compare the values directly from Jakub. Comparing an optional object with a value is supported, see 21 in https://en.cppreference.com/w/cpp/utility/optional/operator_cmp

```suggestion
  if (constMask.value() == 0) {
```

Reply:
Actually, I don't think this does what I need. I need the condition to evaluate to true when there is no value.

Reply:
oh, I misread it, it is not `has_value() && compare`, sorry about that.